server/src/main/java/org/elasticsearch/search/collapse/CollapseBuilder.java

```diff
@@ -203,12 +202,9 @@ public CollapseContext build(QueryShardContext queryShardContext) {
         if (fieldType == null) {
             throw new IllegalArgumentException("no mapping found for `" + field + "` in order to collapse on");
         }
-        if (fieldType instanceof KeywordFieldMapper.KeywordFieldType == false &&
-            fieldType instanceof NumberFieldMapper.NumberFieldType == false) {
-            throw new IllegalArgumentException("unknown type for collapse field `" + field +
-                "`, only keywords and numbers are accepted");
+        if (fieldType.collapseType() == CollapseType.NONE) {
+            throw new IllegalArgumentException("collapse is not supported on this field type");
```

Comment:
Should we keep adding the field name to the error message to make it more actionable? Maybe we also have the actual field type available here to add to the error?

Reply:
@cbuescher Thanks for the feedback, addressed in 621992c